auth0/src/main/java/com/auth0/android/request/DefaultClient.kt

```diff
@@ -18,31 +18,23 @@ import javax.net.ssl.X509TrustManager
 
 /**
  * Default implementation of a Networking Client.
+ *
+ * @param connectTimeout the connection timeout, in seconds, to use when executing requests. Default is ten seconds.
+ * @param readTimeout the read timeout, in seconds, to use when executing requests. Default is ten seconds.
+ * @param defaultHeaders any headers that should be sent on all requests. If a specific request specifies a header with the same key as any header in the default headers, the header specified on the request will take precedence. Default is an empty map.
+ * @param enableLogging whether HTTP request and response info should be logged. This should only be set to `true` for debugging purposes in non-production environments, as sensitive information is included in the logs. Defaults to `false`.
+ * @param sslSocketFactory responsible for creating SSLSockets. Can be set to a custom SSLSocketFactory for unit testing
+ * @param trustManager manages which X509 certificates will be used for authentication.  Can be set to a custom X509TrustManager for unit testing
  */
-public class DefaultClient @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) internal constructor(
-    connectTimeout: Int,
-    readTimeout: Int,
-    private val defaultHeaders: Map<String, String>,
-    enableLogging: Boolean,
-    sslSocketFactory: SSLSocketFactory?,
-    trustManager: X509TrustManager?
+public class DefaultClient constructor(
```

Comment:
We really _don't want_ to expose this constructor that takes the socket factory and trust manager parameters. These were previously exposed internally (for the module) so we could reference and mock them from the tests. Can we roll this visibility back?

Reply:
Thanks for the feedback. We need this exposed for our unit tests to succeed as well as I've noted in the updated Kdoc I made for DefaultClient. Supplying a test ssl socket factory and trustManager is a must as HTTPs is now used by Auth0, or else `PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target` will occur during unit tests. I see no reason for this constructor to be not exposed to developers, do you have any alternatives?

Reply:
I've done more thinking on this and maybe we'll just provide our own internal implementation of NetworkingClient instead of exposing SSLSocketFactory and TrustManager in DefaultClient's constructor. Will update this PR if all goes well with our unit tests.